### PR TITLE
Merge sleep.py into location_encoder.py

### LINKED ISSUE #447: Merge sleep.py into location_encoder.py
All we do in `sleep.py` is train the `LocationEncoder` on simulated data. We should incorporate this as part of the `LocationEncoder` pytorch lightning object to make it more consistent with `BinaryEncoder` and `GalaxyEncoder`.

### PR DESCRIPTION
**Pull request by dereklhansen**
Closes #447

### CODE REVIEW

**Comment by dereklhansen** [comment]
Fixes #537.

**Comment by dereklhansen** [comment]
@ismael-mendoza Currently in the process of retraining the location encoder for `case_studies/sdss_galaxies`, but other than that this is mostly complete.

**Comment by dereklhansen** [comment]
This also fixes #538.

**Comment by ismael-mendoza** [comment]
Thanks @dereklhansen - I'm taking a look now !

File: bliss/models/location_encoder.py [hunk 1]
```diff
@@ -479,3 +702,91 @@ def forward(self, x):
...
+        star_params_log_probs_all_perm[:, i] = (
+            star_params_log_probs_all[:, perm].diagonal(dim1=1, dim2=2)
+            * is_on_array
+            * (1 - true_galaxy_bools)
+        ).sum(1)
+
+    return locs_log_probs_all_perm, star_params_log_probs_all_perm
+
+
+def get_min_perm_loss(
```
Comment (ismael-mendoza) [inline bliss/models/location_encoder.py:742]: I noticed that you made some methods private, I think you can do the same with this one and the next one

File: bliss/models/location_encoder.py [hunk 2]
```diff
@@ -479,3 +702,91 @@ def forward(self, x):
...
+    # view to evaluate all combinations of log_prob.
+    true_params = true_params.view(n_ptiles, 1, max_detections, -1)
+    param_mean = param_mean.view(n_ptiles, max_detections, 1, -1)
+    param_logvar = param_logvar.view(n_ptiles, max_detections, 1, -1)
+
+    sd = (param_logvar.exp() + 1e-5).sqrt()
+    return Normal(param_mean, sd).log_prob(true_params).sum(dim=3)
+
+
+def plot_image_and_locs(
```
Comment (ismael-mendoza) [inline bliss/models/location_encoder.py:781]: Why do you have this empty function?

**Review by ismael-mendoza — APPROVED** [review]
Looks really good @dereklhansen - I just had a couple of questions below.

File: tests/test_encoder_forward.py [hunk 3]
```diff
@@ -1,14 +1,8 @@
-import numpy as np
```
Comment (ismael-mendoza) [inline tests/test_encoder_forward.py:1]: Are you also fixing #531 ?

Comment (dereklhansen) [reply tests/test_encoder_forward.py:1]: Yeah that's correct - I'll link it

File: bliss/models/location_encoder.py (hunk 2 above)
Comment (dereklhansen) [reply bliss/models/location_encoder.py]: Fixed - that was an accident

File: bliss/models/location_encoder.py (hunk 1 above)
Comment (dereklhansen) [reply bliss/models/location_encoder.py]: Done

**Comment by jeff-regier** [comment]
Fantastic!

**Comment by jeff-regier** [comment]
Fantastic!

**Pull request merged**